src/script/conversation/ConversationRepository.ts

```diff
@@ -514,27 +515,69 @@ export class ConversationRepository {
 
   /**
    * Will load all the conversations in memory
-   * @returns all the load conversations from backend merged with the locally stored conversations
+   * @returns all the conversations from backend merged with the locally stored conversations and loaded into memory
    */
   public async loadConversations(): Promise<Conversation[]> {
     const remoteConversationsPromise = this.conversationService.getAllConversations().catch(error => {
       this.logger.error(`Failed to get all conversations from backend: ${error.message}`);
-      return {found: []};
+      return {found: []} as RemoteConversations;
     });
 
     const [localConversations, remoteConversations] = await Promise.all([
       this.conversationService.loadConversationStatesFromDb<ConversationDatabaseData>(),
       remoteConversationsPromise,
     ]);
+    return this.handleRemoteConversationsPromise(remoteConversations, localConversations);
+  }
+
+  /**
+   * Will try to fetch and load all the missing conversations in memory
+   * @returns all the missing conversations freshly fetched from backend appended to the locally stored conversations
+   */
+  public async loadMissingConversations(): Promise<Conversation[]> {
+    const missingConversations = this.conversationState.missingConversations;
+    if (missingConversations.length) {
```

Comment:
I'll go with smth like that:
```suggestion
    if (!missingConversations.length) {
        return this.conversationState.conversations();
    }
    ...rest of code
```

Reply:
Rest of code will be more readable ;)